CitationStyleLanguagePlugin.php

```diff
@@ -0,0 +1,1048 @@
+    }
+
+    public function getTranslatorGroups(int $contextId = 0): array
+    {
+        return $this->getSetting($contextId, 'groupTranslator') ?? [];
+    }
+
+    public function getAuthorGroups(int $contextId = 0): array
+    {
+        return $this->getSetting($contextId, 'groupAuthor') ?? [];
+    }
+
+    public function getChapterAuthorGroups(int $contextId = 0): array
+    {
+        return $this->getSetting($contextId, 'groupChapterAuthor') ?? [];
+    }
+
+    public function getSerialNumber(Publication $publication): array
+    {
+        $serialNumber = [];
+        $publicationFormats = $publication->getData('publicationFormats');
+        /** @var PublicationFormat $publicationFormat */
+        foreach ($publicationFormats as $publicationFormat) {
+            if ($publicationFormat->getIsApproved()) {
+                $identificationCodes = $publicationFormat->getIdentificationCodes();
+                $identificationCodes = $identificationCodes->toArray();
+                foreach ($identificationCodes as $identificationCode) {
+                    $serialNumber[] = htmlspecialchars($identificationCode->getValue());
+                }
+            }
+        }
+        return $serialNumber;
+    }
+
+    protected function addSeriesInformation(stdClass $citationData, Publication $publication): stdClass
+    {
+        $seriesId = $publication->getData('seriesId');
+        if ($seriesId) {
+            /** @var SeriesDAO $seriesDao */
+            $seriesDao = DAORegistry::getDAO('SeriesDAO');
+            if (null !== $seriesDao) {
+                $series = $seriesDao->getById($seriesId);
+                if (null !== $series) {
+                    $citationData->{'collection-title'} = htmlspecialchars(trim($series->getLocalizedFullTitle()));
+                    $citationData->volume = htmlspecialchars($publication->getData('seriesPosition'));
+                    $citationData->{'collection-editor'} = htmlspecialchars($series->getEditorsString());
+                    $onlineISSN = $series->getOnlineISSN();
+                    if (!empty($onlineISSN)) {
+                        $citationData->serialNumber[] = htmlspecialchars($onlineISSN);
+                    }
+                    $printISSN = $series->getPrintISSN();
+                    if (!empty($printISSN)) {
+                        $citationData->serialNumber[] = htmlspecialchars($printISSN);
+                    }
+                }
+            }
+        }
+        return $citationData;
+    }
+
+    protected function getChapter(PKPRequest $request, Publication $publication): ?Chapter
+    {
+        $args = $request->getRequestedArgs();
+
+        if (isset($args[1], $args[2]) && $args[1] === 'chapter') {
+            $key = 2;
+        } elseif (isset($args[1], $args[3], $args[4]) && $args[1] === 'version' && $args[3] === 'chapter') {
+            $key = 4;
+        } else {
+            $key = 0;
```

Comment:
I'd suggest a `return null` right away rather than continuing below. Saves some indentation, and keeps "invalid" code paths as short as possible.